tests/kernel/timer/timer_monotonic/src/main.c

```diff
@@ -8,21 +8,132 @@
 #include <tc_util.h>
 #include <ztest.h>
 
-int test_frequency(void)
+static struct k_timer test_timer;
+static struct k_timer test_periodtimer;
+
+u32_t start_time, stop_time, cycles_spent, nanoseconds_spent,
+	   time_spent[10], cycles_cost[10], pctlist[10];
+int count = 0;
+
+
+
+static void callback_period(struct k_timer *timer)
+{
+	u32_t pct;
+
+	stop_time = k_cycle_get_32();
+	cycles_spent = stop_time - start_time;
+	nanoseconds_spent = (u32_t)k_cyc_to_ns_floor64(cycles_spent);
+	pct = (u64_t)cycles_spent * 100000U / sys_clock_hw_cycles_per_sec();
+	start_time = stop_time;
+	pctlist[count] = pct;
+	time_spent[count] = nanoseconds_spent;
+	cycles_cost[count] = cycles_spent;
+
+	pct =0;
+
+	count++;
+	if (count >= 10) {
+		k_timer_stop(timer);
+	}
+}
+
+static void callback_duration(struct k_timer *timer)
 {
-	u32_t start, end, delta, pct;
+	stop_time = k_cycle_get_32();
+}
+
+int test_periodtimer_frequency(void)
+{
+	int i;
+
+	TC_PRINT("Testing system tick frequency\n");
+
+	k_sleep(1);
+	k_timer_init(&test_periodtimer, callback_period, NULL);
+	start_time = k_cycle_get_32();
+	k_timer_start(&test_periodtimer, 1, 1);
```

Comment:
You want `K_MSEC(1)` here.  This code won't compile against master anymore, the arguments are k_timeout_t now, not integers.